datafusion/sql/tests/cases/plan_to_sql.rs

```diff
@@ -127,7 +127,10 @@ fn roundtrip_statement() -> Result<()> {
             UNION ALL
             SELECT j2_string as string FROM j2
             ORDER BY string DESC
-            LIMIT 10"#
+            LIMIT 10"#,
+            "SELECT id, count(*) over (PARTITION BY first_name ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING), 
```

Comment:
The roundtrip test will fail if `ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING` is not explicitly included. E.g. the datafusion planner generates a non identical plan for the following two SQL queries:

```sql
SELECT id, 
count(*) over (PARTITION BY first_name ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING), 
last_name, 
sum(id) over (PARTITION BY first_name ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING),
first_name from person
```
vs

```sql
SELECT id, 
count(*) over (PARTITION BY first_name), 
last_name, 
sum(id) over (PARTITION BY first_name),
first_name from person
```
While the two plans are not identical `p1!=p2`, I believe the difference is trivial and will actually result in the same computations taking place.

Reply:
> While the two plans are not identical p1!=p2, I believe the difference is trivial and will actually result in the same computations taking place.

That is my understanding as well

We could plausibly simply the resulting plan of the window bounds are the default

Can we also add some tests that have aggregate and window functions? Something like 

```sql
SELECT id, count(distinct id), sum(id) OVER (PARTITION BY first_name) from person

SELECT id, sum(id) OVER (PARTITION BY first_name ROWS 5 PRECEDING ROWS 2 FOLLOWING) from person
```

Reply:
It appears that the datafusion planner does not support mixing aggregate and window functions. It does allow mixing window functions with different WindowSpecs, including some over all rows (which is effectively the same thing as an aggregate function). 

```bash
DataFusion CLI v38.0.0
> create table person (id int, name varchar);
0 row(s) fetched. 
Elapsed 0.001 seconds.
> insert into person values (1, 'a'), (2, 'b'), (3, 'c');
+-------+
| count |
+-------+
| 3     |
+-------+
1 row(s) fetched. 
Elapsed 0.001 seconds.
> select count(distinct id), sum(id) over (partition by name) from person;
Error during planning: Projection references non-aggregate values: Expression person.id could not be resolved from available columns: COUNT(DISTINCT person.id)
> select count(distinct id) over (), sum(id) over (partition by name) from person;
+---------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------+
| COUNT(person.id) ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING | SUM(person.id) PARTITION BY [person.name] ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING |
+---------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------+
| 3                                                                         | 2                                                                                                  |
| 3                                                                         | 1                                                                                                  |
| 3                                                                         | 3                                                                                                  |
+---------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------+
3 row(s) fetched. 
Elapsed 0.002 seconds.
```
I added some tests and made a few changes to correctly support unparsing a SELECT query with multiple different WindowSpecs.

Reply:
The tests look great now -- thanks @devinjdangelo